### PR TITLE
(maint) Adding additional k8s chains to preserve for 1.21.8

### PR DESCRIPTION
**Pull request by abottchen**
Prior to this change, k8s installs of 1.21.8 would fail attempting to
purge these additional chains and rules.  With this update, they are
skipped just as the other k8s rules are.

Prior to this commit, when running from kurl_test with airgap=true:

```
▶  bolt plan run kurl_test::configure_firewall --run-as=root airgap=true targets=firewall-standalone3.c.adam-316219.internal
Starting: plan kurl_test::configure_firewall
Starting: install puppet and gather facts on firewall-standalone3.c.adam-316219.internal
Finished: install puppet and gather facts with 0 failures in 4.36 sec
Starting: apply catalog on firewall-standalone3.c.adam-316219.internal
Finished: apply catalog with 1 failure in 17.87 sec
Finished: plan kurl_test::configure_firewall in 22.24 sec
Failed on firewall-standalone3.c.adam-316219.internal:
  Resources failed to apply for firewall-standalone3.c.adam-316219.internal
    Firewallchain[KUBE-NODE-PORT:filter:IPv4]: change from 'present' to 'absent' failed: Execution of '/sbin/iptables -t filter -X KUBE-NODE-PORT' returned 4: iptables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-NODE-PORT
    Firewallchain[KUBE-MARK-DROP:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-MARK-DROP' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-MARK-DROP
    Firewallchain[KUBE-SERVICES:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-SERVICES' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-SERVICES
    Firewallchain[KUBE-POSTROUTING:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-POSTROUTING' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-POSTROUTING
    Firewallchain[KUBE-FIREWALL:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-FIREWALL' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-FIREWALL
    Firewallchain[KUBE-NODE-PORT:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-NODE-PORT' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-NODE-PORT
    Firewallchain[KUBE-LOAD-BALANCER:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t
nat -X KUBE-LOAD-BALANCER' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-LOAD-BALANCER
    Firewallchain[KUBE-MARK-MASQ:nat:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t nat -X KUBE-MARK-MASQ' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-MARK-MASQ
    Firewallchain[KUBE-FORWARD:filter:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t filter -X KUBE-FORWARD' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-FORWARD
    Firewallchain[KUBE-NODE-PORT:filter:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t filter -X KUBE-NODE-PORT' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-NODE-PORT
    Firewallchain[KUBE-FIREWALL:filter:IPv6]: change from 'present' to 'absent' failed: Execution of '/sbin/ip6tables -t filter -X KUBE-FIREWALL' returned 4: ip6tables v1.8.4 (nf_tables):  CHAIN_USER_DEL failed (Device or resource busy): chain KUBE-FIREWALL
  changed: 53, failed: 11, unchanged: 58 skipped: 0, noop: 0
Failed on 1 target: firewall-standalone3.c.adam-316219.internal
```

After this commit:

```
▶ bolt plan run kurl_test::configure_firewall --run-as=root airgap=true targets=firewall-standalone3.c.adam-316219.internal
Starting: plan kurl_test::configure_firewall
Starting: install puppet and gather facts on firewall-standalone3.c.adam-316219.internal
Finished: install puppet and gather facts with 0 failures in 5.15 sec
Starting: apply catalog on firewall-standalone3.c.adam-316219.internal
Finished: apply catalog with 0 failures in 16.46 sec
```

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -94,6 +94,17 @@
     'WEAVE-NPC-EGRESS:filter:IPv4',
     'WEAVE-NPC-INGRESS:filter:IPv4',
     'WEAVE-NPC:filter:IPv4',
+    'KUBE-NODE-PORT:filter:IPv4',
```
Comment (jpartlow) [inline manifests/init.pp:97]: Except for KUBE-NODE-PORT, which is a table change (addition?) from nat to filter, these appear to be IPv6 additions of existing IPv4 chain skips. Is it that 1.21 added a bunch of IPv6 chains that were previously IPv4 only?

Comment (abottchen) [reply manifests/init.pp:97]: That is what it looks like.  Aside from that one, they are all ipv6.  Everything would make more sense if not for that one ipv4 one hanging out in there.

**Comment by abottchen** [comment]
This is handling the fact that in the 1.21.8 install, they are ipv6 chains out there when there wasn't previously:

1.19.15 build (standalone-stable):

```
# ip6tables -vL -t nat                                                                                                                     
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
```

1.21.8 build (standalone-unstable):

```
# ip6tables -vL -t nat
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-SERVICES  all      any    any     anywhere             anywhere             /* kubernetes service portals */

Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-POSTROUTING  all      any    any     anywhere             anywhere             /* kubernetes postrouting rules */

Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-SERVICES  all      any    any     anywhere             anywhere             /* kubernetes service portals */

Chain KUBE-MARK-DROP (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MARK       all      any    any     anywhere             anywhere             MARK or 0x8000

Chain KUBE-SERVICES (2 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-NODE-PORT  all      any    any     anywhere             anywhere             ADDRTYPE match dst-type LOCAL

Chain KUBE-POSTROUTING (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 RETURN     all      any    any     anywhere             anywhere             mark match ! 0x4000/0x4000
    0     0 MARK       all      any    any     anywhere             anywhere             MARK xor 0x4000
    0     0 MASQUERADE  all      any    any     anywhere             anywhere             /* kubernetes service traffic requiring SNAT */ random-fully

Chain KUBE-FIREWALL (0 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-MARK-DROP  all      any    any     anywhere             anywhere

Chain KUBE-NODE-PORT (1 references)
 pkts bytes target     prot opt in     out     source               destination

Chain KUBE-LOAD-BALANCER (0 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 KUBE-MARK-MASQ  all      any    any     anywhere             anywhere

Chain KUBE-MARK-MASQ (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MARK       all      any    any     anywhere             anywhere             MARK or 0x4000
```

**Review by jpartlow — APPROVED** [review]
:+1:

**Pull request merged**